### PR TITLE
feat(proof): pathfinder_getClassProof endpoint

### PR DESCRIPTION
**Pull request by odesenfans**
Added a new endpoint `pathfinder/v0.1/pathfinder_getClassProof` that returns the Merkle proof of any class hash in the class trie. 

---------------------------

This new endpoint is necessary to run the Starknet OS to completion as we need the full class commitment as input of the OS.

### CODE REVIEW

File: crates/merkle-tree/src/class.rs [hunk 1]
```diff
@@ -71,6 +74,66 @@ impl<'tx> ClassCommitmentTree<'tx> {
...
+            .casm_hash_at(block.into(), sierra)
+            .context("Querying CASM hash")?;
+        let Some(casm) = casm else {
+            return Ok(None);
+        };
+
+        let value = self.tx.class_commitment_leaf(block, &casm)?.map(|x| x.0);
+
+        Ok(value)
+    }
```
Comment (kkovaacs) [inline crates/merkle-tree/src/class.rs:136]: Hmm, this seems a bit strange. How is this actually different from `ClassStorage::leaf()`?

Comment (kkovaacs) [reply crates/merkle-tree/src/class.rs:136]: @odesenfans

Comment (odesenfans) [reply crates/merkle-tree/src/class.rs:136]: It's not, it's an artifact from earlier attempts. Will remove it.

**Comment by HermanObst** [comment]
Hey @kkovaacs! I addressed your comments and update to latest main.
Im getting a spurious clippy error, but the rest looks good :)

**Comment by HermanObst** [comment]
@kkovaacs sorry I had a misconfiguration locally, now everything should be fine :)

**Review by kkovaacs — APPROVED** [review]
LGTM, thanks a lot for this!

Also, please note that the JSON-RPC 0.8.0 specification _will_ have this functionality as part of its `starknet_getProof` method.

**Review by CHr15F0x — APPROVED** [review]

**Pull request merged**